src/FCIQMCProblem.jl

```diff
@@ -0,0 +1,223 @@
+"""
+    SimulationPlan(; starting_step = 1, last_step = 100, walltime = Inf)
+Defines the duration of the simulation. The simulation ends when the `last_step` is reached
+or the `walltime` is exceeded.
+
+See [`FCIQMCProblem`](@ref), [`QMCSimulation`](@ref).
+"""
+Base.@kwdef struct SimulationPlan
+    starting_step::Int = 0
+    last_step::Int = 100
+    walltime::Float64 = Inf
+end
+
+"""
+    FCIQMCProblem(hamiltonian::AbstractHamiltonian; kwargs...)
+Defines a problem to be solved by the FCIQMC algorithm.
+
+# Common keyword arguments and defaults:
+- `time_step = 0.01`: Initial time step size.
+- `last_step = 100`: Controls the number of steps.
+- `targetwalkers = 1_000`: Target for the 1-norm of the coefficient vector.
+- `start_at = starting_address(hamiltonian)`: Define the initial state vector. This can be a
+    single address, a collection of addresses, a single starting vector, or a collection of
+    starting vectors. If multiple starting vectors are passed, their number must match the
+    number of replicas. If (a collection of) [`AbstractDVec`](@ref)(s) is passed, the
+    keyword arguments `style`, `initiator`, and `threading` are ignored.
+- `style = IsDynamicSemistochastic()`: The [`StochasticStyle`](@ref) of the simulation.
+- `initiator = NonInitiator()`: Whether to use initiators. See [`InitiatorRule`](@ref).
+- `threading`: Default is to use multithreading and
+  [MPI](https://juliaparallel.org/MPI.jl/latest/) if multiple threads are available. Set to
+  `true` to force [`PDVec`](@ref) for the starting vector, `false` for serial computation;
+  may be overridden by `start_at`.
+- `reporting_strategy = ReportDFAndInfo()`: How and when to report results, see
+  [`ReportingStrategy`](@ref).
+- `post_step_strategy = ()`: Extract observables (e.g.
+  [`ProjectedEnergy`](@ref)), see [`PostStepStrategy`](@ref).
+- `n_replicas = 1`: Number of synchronised independent simulations.
+- `replica_strategy = NoStats(n_replicas)`: Which results to report from replica
+  simulations, see [`ReplicaStrategy`](@ref).
+
+# Example
+
+```jldoctest
+julia> hamiltonian = HubbardReal1D(BoseFS(1,2,3));
+
+julia> problem = FCIQMCProblem(hamiltonian; targetwalkers = 500, last_step = 100);
+
+julia> simulation = solve(problem);
+
+julia> simulation.success[]
+true
+
+julia> size(DataFrame(simulation))
+(100, 10)
+```
+
+# Further keyword arguments:
+- `starting_step = 1`: Starting step of the simulation.
+- `walltime = Inf`: Maximum time allowed for the simulation.
```

Comment:
```suggestion
- `last_step = 100`: Maximum number of steps for the simulation.
- `walltime = Inf`: Maximum time allowed for the simulation, in seconds.
```
Is there a nicer way to set the time? Setting e.g. 6 days in terms of seconds might be a bit annoying. Maybe one of the packages that provides units?

Reply:
I was thinking that too, but couldn't find how to do it with a quick search. I thought that you might now how to do this nicely 😺

Reply:
`last_step` is already listed under the "Comon keyword arguments". I'm not sure though, whether splitting up the list is a good idea.